.github/workflows/build-test.yml

```diff
@@ -491,43 +491,78 @@ jobs:
         # We're not running CI on macOS for now because it's one less matrix entry to lower the number of runners used,
         # macOS runners are expensive, and we assume that Ubuntu is enough to cover the UNIX case.
         os: [ubuntu-latest]
-        python: [3.8] # Use flaky tests to run against more versions of Python.
+        pythonVersion: [3.8]
+        python: [nonConda, conda]
+        jupyter: [raw, local, remote]
         # integration: Tests with VS Code, Python extension & real Jupyter
         # notebook: Notebook Tests with VS Code, Python extension & real Jupyter
-        test-suite: [integration, notebook, remoteNotebook, nonRawNotebook, notebookWithoutPythonExt, conda, notebookAndWebview]
+        # notebookAndWebview: Tests with VS Code (possible Notebooks), WebViews, Python extension & real Jupyter
+        test-suite: [integration, notebook]
```

Comment:
The types of tests we want, i think this is yet another way to categorize the tests.
Notebook tests, integration tests, webview tests, IPyWidget Tests, etc.
And each of them will run under different scenarios yet again, under python, conda, non-conda, local, remote, raw, etc.

Reply:
The actual output here is a bit confusing to compute. 

So when running remote, 3.8 we exclude integration?
And when running 3.8, conda, remote, we exclude notebook?

Reply:
Can you add a comment explaining what's included/excluded?

Reply:
> So when running remote, 3.8 we exclude integration?

Yes

Basically if we have n*n permuations, i can tell what particular permuation needs to be excluded.
Here I'm saying from all permuations, if there is one that matches OS = Ubuntu, Python 3.8, Jupyter = Remote and test-suite = integration, then exclude that item from the list.

Reply:
Will add comments.